### PR TITLE
[mysql-connector-cpp] Fix incorrect usage of MAYBE_UNUSED_VARIABLES

### LINKED ISSUE #32356: [mysql-connector-cpp] Build failure
Package: mysql-connector-cpp[core,jdbc]:arm64-osx -> 8.0.32

**Host Environment**

- Host: arm64-osx
- Compiler: AppleClang 14.0.3.14030022
-    vcpkg-tool version: 2999-12-31-unknownhash
    vcpkg-scripts version: 64adda19c 2023-07-02 (7 hours ago)

**To Reproduce**

`vcpkg install --enforce-port-checks --allow-unsupported`

**Failure logs**

```
-- Using cached mysql-mysql-connector-cpp-8.0.32.tar.gz.
-- Extracting source /Users/leanderSchulten/git_projekte/vcpkg/downloads/mysql-mysql-connector-cpp-8.0.32.tar.gz
-- Applying patch fix-static-build8.patch
-- Applying patch export-targets.patch
-- Applying patch dependencies.patch
-- Using source at /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/mysql-connector-cpp/src/8.0.32-0039c1a6e0.clean
-- Configuring arm64-osx
-- Building arm64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /Users/leanderSchulten/git_projekte/vcpkg/downloads/tools/cmake-3.25.1-osx/cmake-3.25.1-macos-universal/CMake.app/Contents/bin/cmake --build . --config Debug --target install -- -v -j11
    Working Directory: /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/mysql-connector-cpp/arm64-osx-dbg
    See logs for more information:
      /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/mysql-connector-cpp/install-arm64-osx-dbg-err.log

Call Stack (most recent call first):
  vcpkg_installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  vcpkg_installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/mysql-connector-cpp/portfile.cmake:45 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)



```
<details><summary>/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/mysql-connector-cpp/install-arm64-osx-dbg-err.log</summary>

```
ninja: error: 'MYSQL_LIB-NOTFOUND', needed by 'jdbc/libconnector-jdbc-deps.dylib', missing and no known rule to make it
```
</details>

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
  "name": "lichtsteuerung",
  "version-string": "2.0 $.0",
  "dependencies": [
    {
      "name": "mysql-connector-cpp",
      "default-features": false,
      "features": [
        "jdbc"
      ]
    }
  ],
  "$overrides": [
    {
      "name": "capstone",
      "version": "4.0.1-120373dc"
    }
  ],
  "$builtin-baseline": "338f99eed499b6df1628829e40695b396a70a1f7"
}

```
</details>

### PR DESCRIPTION
**Pull request by kotori2**
Fixes #32356

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by jimwang118** [comment]
I get the following issue while testing usage.
```
Error		CMake Error at F:/mysql-connector-cpp/scripts/buildsystems/vcpkg.cmake:855 (_find_package):
  Could not find a package configuration file provided by
  "unofficial-mysql-connector-cpp" with any of the following names:

    unofficial-mysql-connector-cppConfig.cmake
    unofficial-mysql-connector-cpp-config.cmake

  Add the installation prefix of "unofficial-mysql-connector-cpp" to
  CMAKE_PREFIX_PATH or set "unofficial-mysql-connector-cpp_DIR" to a
  directory containing one of the above files.  If
  "unofficial-mysql-connector-cpp" provides a separate development package or
  SDK, be sure it has been installed.		F:/mysql-connector-cpp/scripts/buildsystems/vcpkg.cmake	855	
```

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by kotori2** [comment]
@jimwang118 It does work for me with triplet `x64-windows-static` and `arm64-macos`. I've attached a simple test case to show it works. 
[untitled1.zip](https://github.com/microsoft/vcpkg/files/11965394/untitled1.zip)

File: ports/mysql-connector-cpp/portfile.cmake [hunk 1]
```diff
@@ -36,10 +36,13 @@ vcpkg_cmake_configure(
         -DSTATIC_MSVCRT=${STATIC_MSVCRT}
         -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
         -DWITH_JDBC=${WITH_JDBC}  # the following variables are only used by jdbc
-    MAYBE_UNUSED_VARIABLES  # and they are windows only
         -DMYSQL_INCLUDE_DIR="${CURRENT_INSTALLED_DIR}/include/mysql"
-        -DMYSQL_LIB_DIR=${CURRENT_INSTALLED_DIR}
-        -DWITH_BOOST=${CURRENT_INSTALLED_DIR}
+        -DMYSQL_LIB_DIR="${CURRENT_INSTALLED_DIR}"
```
Comment (dg0yt) [inline ports/mysql-connector-cpp/portfile.cmake:40]: Not portable.
```suggestion
        "-DMYSQL_LIB_DIR=${CURRENT_INSTALLED_DIR}"
```

**Comment by jimwang118** [comment]
> @jimwang118 It does work for me with triplet `x64-windows-static` and `arm64-macos`. I've attached a simple test case to show it works. [untitled1.zip](https://github.com/microsoft/vcpkg/files/11965394/untitled1.zip)

The test with the example you provided is correct.

File: ports/mysql-connector-cpp/portfile.cmake (hunk 1 above)
Comment (kotori2) [reply ports/mysql-connector-cpp/portfile.cmake:40]: Will it goes wrong if we have a space in the `CURRENT_INSTALLED_DIR`?

Comment (dg0yt) [reply ports/mysql-connector-cpp/portfile.cmake]: Not with correct quotes.

Comment (dg0yt) [reply ports/mysql-connector-cpp/portfile.cmake]: (... and quotes are missing in more spots, but I don't care that much about this particular port.)

Comment (kotori2) [reply ports/mysql-connector-cpp/portfile.cmake]: > (... and quotes are missing in more spots, but I don't care that much about this particular port.)

I did fixed them in 451a337 :D

**Comment by dg0yt** [comment]
> > @jimwang118 It does work for me with triplet `x64-windows-static` and `arm64-macos`. I've attached a simple test case to show it works. [untitled1.zip](https://github.com/microsoft/vcpkg/files/11965394/untitled1.zip)
> 
> The test with the example you provided is correct.

No, everything is incorrect. It simply doesn't provide the proper unofficial package name. But this is not a regression from this PR.

**Comment by jimwang118** [comment]
> I get the following issue while testing usage.
> 
> ```
> Error		CMake Error at F:/mysql-connector-cpp/scripts/buildsystems/vcpkg.cmake:855 (_find_package):
>   Could not find a package configuration file provided by
>   "unofficial-mysql-connector-cpp" with any of the following names:
> 
>     unofficial-mysql-connector-cppConfig.cmake
>     unofficial-mysql-connector-cpp-config.cmake
> 
>   Add the installation prefix of "unofficial-mysql-connector-cpp" to
>   CMAKE_PREFIX_PATH or set "unofficial-mysql-connector-cpp_DIR" to a
>   directory containing one of the above files.  If
>   "unofficial-mysql-connector-cpp" provides a separate development package or
>   SDK, be sure it has been installed.		F:/mysql-connector-cpp/scripts/buildsystems/vcpkg.cmake	855	
> ```

I use VS2022 to create a cmake project, and then use `find_package(unofficial-mysql-connector-cpp CONFIG REQUIRED)` in `cmakelists.txt` and the above error will appear.

**Comment by kotori2** [comment]
@jimwang118 Still can't reproduce... 
image
image

**Comment by dg0yt** [comment]
> No, everything is incorrect. It simply doesn't provide the proper unofficial package name. But this is not a regression from this PR.

Sorry, this was wrong. What @jimwang118 could do is configuring with `CMAKE_FIND_DEBUG_MODE` set to 1.

**Comment by jimwang118** [comment]
> @jimwang118 Still can't reproduce... image image

There is no problem. I have a problem with configuring the `cmakesetting.json` file. After modifying the configuration, the test result is correct.

**Pull request merged**